### DISCUSSION TITLE
selectedDeliveryOption is always null in the input to function

### DISCUSSION THREAD

**Discussion by SaeedYasin** [Shopify/function-examples#201] [Feedback: Delivery Customization API]
Here is the checkout, where I have tried to select a delivery option and reload checkout, it does seem to remember what delivery option I selected last time.
<img width="600" alt="Screenshot 2023-03-28 at 5 20 54 PM" src="https://user-images.githubusercontent.com/26478201/228190792-98a927c9-82f3-4062-9147-6af1654000ad.png">

But selectedDeliveryOption in the input to the function is always null, no matter what I try, it is never set to any value other than null.
<img width="680" alt="Screenshot 2023-03-28 at 5 20 39 PM" src="https://user-images.githubusercontent.com/26478201/228190730-46ee4116-0c30-49b6-a35d-978fbffa3a35.png">

Is this a known issue? or am I doing it wrong some how? 👀

**Comment by SaeedYasin** [comment]
This only happens in delivery customizations, but works fine in payment customizations.

**Comment by vixnguyen** [comment]
I'm getting `deliveryGroups` in my discount function, and it always returns empty array. The same [here ](https://community.shopify.com/c/shopify-apps/shopify-functions/m-p/1945548/highlight/true#M59812)

**Comment by dhanush-tnva** [comment]
Hey, 

Any solutions to this problem ?

**Comment by aldoalprak** [comment]
is there any update on this? the bug still exists until now

**Comment by Dhanush-git** [comment]
I had a conversation with someone at Shopify, and they explained that the delivery **customization function operates even before the user selects a delivery option**, which is why it returns null. This makes sense, as it’s necessary to display only the relevant delivery options.

I suggest they update the documentation to clarify this and **remove the mention of the selected delivery option**, as it is leading to confusion among developers.
